### PR TITLE
Configure Dependabot to open PRs on staging branch

### PR DESCRIPTION
**Pull request by howardwu**
## Motivation

Configure Dependabot to open PRs on staging branch

### CODE REVIEW

**Comment by howardwu** [comment]
@ljedrz @niklaslong it appears there's an unrelated failure affecting this PR's CI

**Review by weikengchen — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ updates:
   schedule:
     interval: daily
     time: "10:00"
+  target_branch: "staging"
```
Comment (ljedrz) [inline .github/dependabot.yml:8]:
```suggestion
  target-branch: "staging"
```

`dependabot`'s docs suggest that this should be `target-branch` instead: https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#target-branch

**Review by ljedrz — CHANGES_REQUESTED** [review]
I think we need to change the key from an underscore to a dash, as I commented on the diff; other than that it's a desirable change :+1:.

**Comment by niklaslong** [comment]
Re-running the CI jobs fixes both failures in `snarkos-testing` and `rust-stable`.

**Review by ljedrz — APPROVED** [review]

**Review by niklaslong — APPROVED** [review]

**Pull request merged**